pandas/core/internals.py

```diff
@@ -5434,6 +5443,13 @@ def is_uniform_join_units(join_units):
         len(join_units) > 1)
 
 
+def is_uniform_reindex(join_units):
+    return (
+        # TODO: should this be ju.block.can_hold_na?
+        all(ju.block and ju.block.is_extension for ju in join_units) and
+        len(set(ju.block.dtype.name for ju in join_units)) == 1
```

Comment:
Is this checking of only the name potentially dangerous for parameterized dtypes like categorical?

Reply:
Yes, I meant to go back and check that :) I haven't figured out how to best handle it, but this the current version needs to be improved.